### PR TITLE
HOTFIX: fix precommit

### PR DESCRIPTION
**Pull request by raphael-speyer-da**
It seems the precommit entries are run from `$GIT_DIR`, so this adapts to the fact that the code has been moved under the `sdk/` subdir.

We can now run `pre-commit run --all-files` successfully. As things stand `javafmt` still fails, but that's because the code is currently not formatted as this tool would do it. This is code that comes in from `canton`, so we should probably align the formatting on that side and/or exclude this code from formatting.

### CODE REVIEW

File: sdk/package.json
Comment (raphael-speyer-da) [inline sdk/package.json:1]: This file is now a symlink from the dir above

Comment (raphael-speyer-da) [inline sdk/package.json:1]: This file was moved up a dir and turned into a symlink from there.

**Comment by garyverhaegen-da** [comment]
Not having package.json at the top level is more important to me than having the precommit script work. (Or exist at all.)

**Comment by raphael-speyer-da** [comment]
> Not having package.json at the top level is more important to me than having the precommit script work. (Or exist at all.)

Okay, no need to move `package.json` any more.

File: sdk/.pre-commit-config.yaml [hunk 1]
```diff
@@ -30,14 +30,14 @@ repos:
...
-      entry: "bazel run //:buildifier-pre-commit -- -mode=fix -v=true"
+      entry: "bash -c 'cd sdk/; bazel run //:buildifier-pre-commit -- -mode=fix -v=true'"
       types: [bazel]
     - id: pprettier
       name: pprettier
       language: system
       require_serial: true
       # NB: we need to pass a single argument which ends up in $0 when invoking bash -c
-      entry: "bash -c 'yarn install --silent && yarn run pprettier --write \"$@\"' bash"
+      entry: "bash -c 'yarn install --silent && files=(\"$@\") && yarn run --cwd=sdk pprettier --write ${files[@]##sdk/}' bash"
```
Comment (raphael-speyer-da) [inline sdk/.pre-commit-config.yaml:40]: * `files=(\"$@\")` creates a bash array out of the arguments (the discovered `.ts` and `.tsx` files).
* ` --cwd=sdk` makes yarn run in the `sdk/` dir, which is where the `package.json` file lives.
* `${files[@]##sdk/}` is an operation which removes the `sdk/` prefix from each element of the array.

**Review by garyverhaegen-da — APPROVED** [review]
Thanks!

**Comment by garyverhaegen-da** [comment]
I'll get this merged once Windows nodes work again.

**Pull request merged**